app/utils/PHProvider.tsx

```diff
@@ -0,0 +1,23 @@
+'use client';
+// import posthog from 'posthog-js'
+import { PostHogProvider } from 'posthog-js/react';
+import { ReactNode, useEffect, useState } from 'react';
+
+export function PHProvider({
+  children,
+}: Readonly<{
+  children: ReactNode
+}>) {
+  const [isClient, setIsClient] = useState(false);
+  useEffect(() => {
+    if (window !== undefined) {
+      setIsClient(true);
+    }
+  }, []);
+
+  if (!isClient) {
+    return <>{children}</>;
+  }
+
+  return <PostHogProvider client={window.posthog}>{children}</PostHogProvider>;
```

Comment:
I have added window.posthog here since the google tag manager embeds the posthog code in window

https://www.googletagmanager.com/gtm.js?id=GTM-N8LG5FMQ